### ISSUE TITLE
loss_function='MAPE' breaks the prediction

### ISSUE THREAD

**Issue by slivka83** [catboost/catboost#1049]
Problem: loss_function='MAPE' breaks the prediction.
catboost version: 0.18
Operating System: win10
CPU: 2700X AMD
GPU: 1080 Ti

When I set loss_function='MAPE', the prediction breaks.
```
from catboost import CatBoostRegressor
from sklearn.datasets import make_regression

X, y = make_regression(n_features=4, random_state=0)

cat = CatBoostRegressor(
    random_seed = 0,
    loss_function='MAPE',
    task_type = 'GPU'
)

cat.fit(X, y, verbose=100)

print(cat.predict(X))
```
Output:
[-0.50954083  1.75879133 -0.30212357 -0.2024278   0.48867573 -0.35168457
 -0.1707202  -0.30207277 -0.21107998  0.09352795 -0.57918989 -0.03051643
 -0.30995589 -0.31928853 -0.25451159 -0.14208117 -0.29449031 -0.176887
  0.09035442 -0.9967452   0.6619211   0.15376783 -0.05623467  0.28612917
 -0.24599992 -0.42766109  0.11618973 -0.44268044  0.38049769  0.51400905
  0.31298011  1.58461204  0.65129645  0.15588823 -0.47914321  0.45968965
 -0.60523516  0.27460327 -0.24720762 -0.25359377  0.43629071 -0.42985261
  0.196212    0.40976468 -0.22220335 -0.28804626  0.10452159 -0.11516057
 -0.43264386  0.16228754 -0.20642639  0.28182217  0.2737032   0.16804635
 -0.20726307 -0.41797646  0.27831955 -0.4470549  -0.3697947  -0.02817013
  0.58189431  0.44387035  0.12134547  0.0033957  -0.05080942 -0.11577619
 -0.24518521  0.2380991   0.0054664  -0.29447758  0.17677258 -0.46538358
 -0.23650918 -0.60060311 -0.30394749  0.15282923 -0.42326767 -0.20894282
 -0.29866385 -0.26556007 -0.44746338 -0.62030356 -0.44212303 -0.2736055
 -0.31168719 -0.24684543  0.26430054  0.36793473 -0.18490792  0.16918891
 -0.26130316 -0.10702055 -0.28072999 -0.29435111  0.1224026  -0.20235984
 -0.57723527 -0.81178908 -0.08444301 -0.2810646 ]

When default loss_function:
```
cat = CatBoostRegressor(
    random_seed = 0,
    task_type = 'GPU'
)

cat.fit(X, y, verbose=100)

print(cat.predict(X))
```
Output:
[ -39.28033274    3.43609111  -83.12582161  -36.82084191   33.5601792
  -37.68219423   42.40785187 -119.79732787  -52.02199758  109.50237902
 -117.88383553   95.79343728  -79.61345129  -41.61094593 -152.96254887
  -60.872345   -123.16551755 -165.50667459   20.00291066  -11.73762142
   46.28123038   98.61265844  137.10613531   96.18421529 -104.68325357
  -32.38198063  282.33368733  -62.53496734   59.46395189   67.58367808
  123.542776     10.71650631   12.86872335  127.01375566 -117.23444883
   88.59164231   15.40909896  109.58026324 -209.86485698 -267.9141543
  118.78392494  -32.21124833   98.1732494   105.81400817 -111.25715379
  -40.43695291  117.37267141 -177.24909844  -56.92621069  109.17892832
 -132.94873384   50.88852617  157.36407411  295.88195135 -168.17834477
  -59.30795042   53.17354119  -58.73239884  -78.40170724   69.96047914
   10.67272478  104.47469668  302.35625185   47.40169511   92.39832878
  105.47200503 -234.26908356   82.12914917  100.52413374  -51.62022542
  129.03988545  -10.04911046 -232.77394387  -19.93882009  -54.65705414
  161.76501868  -60.43905224  -23.20730159 -122.77772255  -82.01832318
   -5.23463334  -26.37461449  -62.05417034 -112.66976874  -37.25035492
  -29.95026429   12.71166442  155.78570566 -204.96156033  170.89051143
 -122.33506787  117.89355815 -154.40665628  -44.97260419  269.43099015
 -247.15739064  -14.80384957  -10.2649668    59.70649853 -103.80268756]

**Comment by LyzhinIvan** [comment]
Hi, @slivka83 ! Because of the nature of MAPE loss function (gradient for each point equals to +1/target or -1/target), fitting with gradient method is very slow (in terms of loss function change per iteration) when target values are large. We have implemented Exact leaf estimation method for MAPE for CPU that is much better and used by default. Currently it's not available for GPU. So I can recommend you to train your model with CPU or use default loss (RMSE) with GPU.

**Comment by annaveronika** [comment]
I hope it has helped, let us know if you have more questions.
